src/coreclr/jit/gentree.cpp

```diff
@@ -18500,6 +18500,7 @@ bool Compiler::gtStoreDefinesField(
 //    tree -- tree to find handle for
 //    pIsExact   [out] -- whether handle is exact type
 //    pIsNonNull [out] -- whether tree value is known not to be null
+//    vnBased          -- can we rely on value numbering to find the handle?
```

Comment:
Introduced `fgVnValid` for it, going to think about using it in various asserts.

Reply:
`vnStore != nullptr` could be used as a check, it would avoid an additional boolean.

Reply:
> `vnStore != nullptr` could be used as a check, it would avoid an additional boolean.

@SingleAccretion does vnStore becomes null once VN are stale?

Reply:
Looks like it's not. It's only to null set in ResetOptAnnotations (which is JitOptRepeat specific)

Reply:
Yes, but you could null it out just like you're setting `fgVnValid` to `false`. It would "self-validate" that nothing uses it that way as well.

Reply:
Done (with a small fix in gtExtractSideEffList which was called in late phases)